lib/community/pages/create_post_page.dart

```diff
@@ -32,51 +41,102 @@ import 'package:thunder/utils/instance.dart';
 class CreatePostPage extends StatefulWidget {
   final int? communityId;
   final CommunityView? communityView;
-  final void Function(DraftPost? draftPost)? onUpdateDraft;
-  final DraftPost? previousDraftPost;
 
-  // used create post from action sheet
+  /// Whether or not to pre-populate the post with the [title], [text], [image] and/or [url]
+  final bool? prePopulated;
```

Comment:
I was wondering if we might want to think about consolidating all the different ways we can prepopulate a post. Right now I think we have drafts, edits, and shares, each with different things that are saved/restored. (For example, is language selection saved with drafts?) Maybe we can have a wrapper class around a `Post/View/Media/whatever`, so that it can contain every possible field, and then an enum saying what kind of thing it is. Just a thought.

Reply:
Thats a good idea! I think we could do this in a different PR since this one already contains a pretty substantial change if thats all good with you.

Reply:
Of course!